tracer/test/Datadog.Trace.Tests/Configuration/TracerSettingsTests.cs

```diff
@@ -1234,6 +1234,48 @@ public void InitializeHeaderTag(string tagName, bool headerTagsNormalizationFixE
             finalTagName.Should().Be(expectedTagName);
         }
 
+        [Theory]
+        [InlineData(null, null, "555")]
+        [InlineData(null, "400", "400")]
+        [InlineData("444", null, "444")]
+        [InlineData("444", "424", "444")]
+        public void ValidateServerErrorStatusCodes(string newServerErrorKeyValue, string deprecatedServerErrorKeyValue, string expectedServerError)
+        {
+            const string httpServerErrorStatusCodes = "DD_TRACE_HTTP_SERVER_ERROR_STATUSES";
+            const string deprecatedHttpServerErrorStatusCodes = "DD_HTTP_SERVER_ERROR_STATUSES";
+
+            var source = CreateConfigurationSource(
+                (httpServerErrorStatusCodes, newServerErrorKeyValue),
+                (deprecatedHttpServerErrorStatusCodes, deprecatedServerErrorKeyValue));
+
+            var errorLog = new OverrideErrorLog();
+            var settings = new TracerSettings(source, NullConfigurationTelemetry.Instance, errorLog);
+            var result = settings.HttpServerErrorStatusCodes;
+
+            Assert.True(result[int.Parse(expectedServerError)]);
+        }
+
+        [Theory]
+        [InlineData(null, null, "444")]
+        [InlineData(null, "500", "500")]
+        [InlineData("555", null, "555")]
+        [InlineData("555", "525", "555")]
+        public void ValidateClientErrorStatusCodes(string newClientErrorKeyValue, string deprecatedClientErrorKeyValue, string expectedClientError)
```

Comment:
nipticky:

To me this test data reads strangely - `null`, `null`, `"555"`
Where the reality is the "expected` server errors is actually the default range - not just a single value.
Could we maybe assert on that range?

Reply:
https://github.com/DataDog/dd-trace-dotnet/pull/6095/files/7fa9adde56eb12badda28c82929e7f14d56b8d82#r1781423631(Will look into it).

Reply:
Added testing for default range in: https://github.com/DataDog/dd-trace-dotnet/pull/6095/commits/416297a5b77ded91243d5643258ea159bf81b461